crates/frontend/src/contextmenu/mod.rs

```diff
@@ -1,17 +1,179 @@
 use std::rc::Rc;
 use futures_signals::signal::Mutable;
+use dominator::{Dom, html, clone, events};
+use wasm_bindgen::JsValue;
+use crate::{vfs::{Directory, File}, DEFAULT_DIRECTORY_MODE, DEFAULT_FILE_MODE};
 
-pub struct ContextMenuState {
-    pub show_menu: Mutable<bool>,
-    pub menu_position: Mutable<(i32, i32)>,
+#[derive(Clone, Copy)]
+pub enum Menutype {
+    File, 
+    Directory
+}
+
+pub struct ContextMenu {
+    // visibility and position of contextmenu
+    pub show: Mutable<bool>,
+    pub position: Mutable<(i32, i32)>,
+    // to split the contextmenu separately for file and folder
+    pub menu_type: Mutable<Option<Menutype>>,
+    // to see which file or folder is clicked
+    pub target_object: Mutable<Option<Rc<Directory>>>
```

Comment:
Can the target object also be a `Mutable<Option<Rc<File>>>`?

Reply:
Yes, it should ideally also be a file since it will allow us to track which object we will be modifying. I think we can approach this using `Enum` so that we can have both File and Directory.